Computational Biology Labs 2022-2023
===================================
Lab 2: Sequence Alignment
=========================

In this Lab, we will learn about sequence alignment, and how to use the Needleman-Wunsch algorithm and the Smith-Waterman algorithm to align two sequences.

During this Lab, the Smith-Waterman algorithm was implemented in Python and used to align two sequences.

Both of these algorithms are dynamic programming algorithms for finding local alignments between two strings. They are used in bioinformatics to find regions of similarity between biological sequences. They are commonly used for finding regions of similarity between DNA, RNA, or protein sequences.

The Smith-Waterman algorithm is used to find the best local alignment between two sequences. The Needleman-Wunsch algorithm is used to find the best global alignment between two sequences.


The alignment of two sequences is a very important task in bioinformatics. It is used to compare two sequences, and to find similarities between them. It is also used to find the best match between two sequences, and to find the best match between a sequence and a database of sequences.

